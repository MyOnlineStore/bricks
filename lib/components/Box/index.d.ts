import { FunctionComponent } from 'react';
import { TrblType } from '../../utility/trbl';
import { OffsetShorthandType } from '../../types/OffsetType';
declare type PropsType = JSX.IntrinsicElements['div'] & {
    justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around' | 'space-evenly';
    alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
    alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around';
    direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
    inline?: boolean;
    height?: string;
    width?: string;
    margin?: TrblType | OffsetShorthandType;
    padding?: TrblType | OffsetShorthandType;
    maxHeight?: string;
    minHeight?: string;
    maxWidth?: string;
    minWidth?: string;
    wrap?: boolean;
    grow?: number;
    shrink?: number;
    basis?: string;
    order?: number;
    alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
    position?: 'static' | 'relative' | 'fixed' | 'absolute';
    top?: string;
    right?: string;
    bottom?: string;
    left?: string;
};
declare const Box: FunctionComponent<PropsType>;
export { PropsType };
export default Box;
