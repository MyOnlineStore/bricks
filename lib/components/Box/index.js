!function(n,e){"object"==typeof exports&&"object"==typeof module?module.exports=e(require("react"),require("styled-components")):"function"==typeof define&&define.amd?define("bricks",["react","styled-components"],e):"object"==typeof exports?exports.bricks=e(require("react"),require("styled-components")):n.bricks=e(n.react,n["styled-components"])}("undefined"!=typeof self?self:this,function(n,e){return function(n){var e={};function t(r){if(e[r])return e[r].exports;var i=e[r]={i:r,l:!1,exports:{}};return n[r].call(i.exports,i,i.exports,t),i.l=!0,i.exports}return t.m=n,t.c=e,t.d=function(n,e,r){t.o(n,e)||Object.defineProperty(n,e,{configurable:!1,enumerable:!0,get:r})},t.n=function(n){var e=n&&n.__esModule?function(){return n.default}:function(){return n};return t.d(e,"a",e),e},t.o=function(n,e){return Object.prototype.hasOwnProperty.call(n,e)},t.p="",t(t.s=3)}([function(e,t){e.exports=n},function(n,e,t){"use strict";Object.defineProperty(e,"__esModule",{value:!0}),t.d(e,"css",function(){return u}),t.d(e,"injectGlobal",function(){return a}),t.d(e,"keyframes",function(){return f}),t.d(e,"ThemeProvider",function(){return c}),t.d(e,"withProps",function(){return d});var r=t(2),i=(t.n(r),r),o=i.default,u=i.css,a=i.injectGlobal,f=i.keyframes,c=i.ThemeProvider;function d(n){return n}e.default=o},function(n,t){n.exports=e},function(n,e,t){"use strict";Object.defineProperty(e,"__esModule",{value:!0});var r=t(0),i=t.n(r),o=t(5),u=this&&this.__assign||Object.assign||function(n){for(var e,t=1,r=arguments.length;t<r;t++)for(var i in e=arguments[t])Object.prototype.hasOwnProperty.call(e,i)&&(n[i]=e[i]);return n},a=this&&this.__rest||function(n,e){var t={};for(var r in n)Object.prototype.hasOwnProperty.call(n,r)&&e.indexOf(r)<0&&(t[r]=n[r]);if(null!=n&&"function"==typeof Object.getOwnPropertySymbols){var i=0;for(r=Object.getOwnPropertySymbols(n);i<r.length;i++)e.indexOf(r[i])<0&&(t[r[i]]=n[r[i]])}return t};e.default=function(n){var e=n.order,t=n.direction,r=n.wrap,f=n.height,c=n.width,d=n.maxHeight,l=n.minHeight,s=n.maxWidth,h=n.minWidth,p=(n.ref,a(n,["order","direction","wrap","height","width","maxHeight","minHeight","maxWidth","minWidth","ref"])),v=u({},p,{flexWrap:r,elementHeight:f,elementWidth:c,elementMaxHeight:d,elementMinHeight:l,elementMaxWidth:s,elementMinWidth:h,flexDirection:t,flexOrder:e});return n.inline?i.a.createElement(o.b,u({},v),n.children):i.a.createElement(o.a,u({},v),n.children)}},,function(n,e,t){"use strict";t.d(e,"a",function(){return u}),t.d(e,"b",function(){return a});var r,i=t(1),o=this&&this.__makeTemplateObject||function(n,e){return Object.defineProperty?Object.defineProperty(n,"raw",{value:e}):n.raw=e,n},u=Object(i.withProps)(i.default.div)(r||(r=o(["\n    box-sizing: border-box;\n    display: ",";\n    height: ",";\n    width: ",";\n    max-height: ",";\n    min-height: ",";\n    max-width: ",";\n    min-width: ",";\n    flex-wrap: ",";\n    flex-direction: ",";\n    justify-content: ",";\n    align-items: ",";\n    align-content: ",";\n    margin: ",";\n    padding: ",";\n    flex-grow: ",";\n    flex-shrink: ",";\n    flex-basis: ",";\n    order: ",";\n    align-self: ",";\n    ",";\n    ",";\n    ",";\n    ",";\n    ",";\n"],["\n    box-sizing: border-box;\n    display: ",";\n    height: ",";\n    width: ",";\n    max-height: ",";\n    min-height: ",";\n    max-width: ",";\n    min-width: ",";\n    flex-wrap: ",";\n    flex-direction: ",";\n    justify-content: ",";\n    align-items: ",";\n    align-content: ",";\n    margin: ",";\n    padding: ",";\n    flex-grow: ",";\n    flex-shrink: ",";\n    flex-basis: ",";\n    order: ",";\n    align-self: ",";\n    ",";\n    ",";\n    ",";\n    ",";\n    ",";\n"])),function(n){return n.inline?"inline-flex":"flex"},function(n){var e=n.elementHeight;return void 0!==e?e:""},function(n){var e=n.elementWidth;return void 0!==e?e:""},function(n){var e=n.elementMaxHeight;return void 0!==e?e:""},function(n){var e=n.elementMinHeight;return void 0!==e?e:""},function(n){var e=n.elementMaxWidth;return void 0!==e?e:""},function(n){var e=n.elementMinWidth;return void 0!==e?e:""},function(n){var e=n.flexWrap;return void 0!==e&&e?"wrap":""},function(n){var e=n.flexDirection;return void 0!==e?e:""},function(n){var e=n.justifyContent;return void 0!==e?e:""},function(n){var e=n.alignItems;return void 0!==e?e:""},function(n){var e=n.alignContent;return void 0!==e?e:""},function(n){var e=n.margin;return void 0!==e?e.top+" "+e.right+" "+e.bottom+" "+e.left:""},function(n){var e=n.padding;return void 0!==e?e.top+" "+e.right+" "+e.bottom+" "+e.left:""},function(n){var e=n.grow;return e||0},function(n){var e=n.shrink;return void 0!==e?e:1},function(n){var e=n.basis;return e||"auto"},function(n){var e=n.flexOrder;return e||0},function(n){var e=n.alignSelf;return e||""},function(n){var e=n.position;return void 0!==e?"position: "+e:""},function(n){var e=n.top;return void 0!==e?"top: "+e:""},function(n){var e=n.right;return void 0!==e?"right: "+e:""},function(n){var e=n.bottom;return void 0!==e?"bottom: "+e:""},function(n){var e=n.left;return void 0!==e?"left: "+e:""}),a=u.withComponent("span")}])});
//# sourceMappingURL=index.js.map