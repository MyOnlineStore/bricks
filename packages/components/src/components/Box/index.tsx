import React, { FunctionComponent } from 'react';
import { OffsetShorthandType } from '../../types/OffsetType';
import { FlexProps, BoxProps, flex, box } from '../../utility/box';
import styled from '../../utility/styled';
import { TrblType } from '../../utility/trbl';

const StyledBox = styled.div<FlexProps & BoxProps>`
    ${flex}
    ${box}
`;

/**  These props exist for backward compatibility */
type PropsType = BoxProps &
    FlexProps &
    JSX.IntrinsicElements['div'] & {
        as?: keyof JSX.IntrinsicElements;
        height?: string;
        width?: string;
        margin?: TrblType | OffsetShorthandType;
        padding?: TrblType | OffsetShorthandType;
        maxHeight?: string;
        minHeight?: string;
        maxWidth?: string;
        minWidth?: string;
        zIndex?: number;
        position?: 'static' | 'relative' | 'fixed' | 'absolute';
        top?: string;
        right?: string;
        bottom?: string;
        left?: string;
        order?: number;
        direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
        justifyContent?:
            | 'flex-start'
            | 'flex-end'
            | 'center'
            | 'stretch'
            | 'space-between'
            | 'space-around'
            | 'space-evenly';
        alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline';
        alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around';
        inline?: boolean;
        wrap?: boolean;
        grow?: number;
        shrink?: number;
        basis?: string;
        alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
    };

const Box: FunctionComponent<PropsType> = props => {
    const { ref, ...filteredProps } = props;
    const mappedProps = Object.keys(filteredProps).reduce((acc, key) => {
        return {
            ...acc,
            [`$${key.replace('$', '')}`]: filteredProps[key as keyof typeof filteredProps],
        };
        // tslint:disable-next-line: no-any
    }, {} as any);

    return (
        // tslint:disable-next-line: no-any
        <StyledBox ref={ref} as={(props.inline ? 'span' : 'div') as any} {...mappedProps}>
            {props.children}
        </StyledBox>
    );
};

export { PropsType };
export default Box;
