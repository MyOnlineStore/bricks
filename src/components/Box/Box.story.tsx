import { boolean, select } from '@storybook/addon-knobs/react';
import { storiesOf } from '@storybook/react';
import React from 'react';
import styled from 'styled-components';
import Spacer from '../Spacer/Spacer.style';
import Text from '../Text';
import Box from './Box.style';

const Wrapper = styled.div`
    border: 1px dashed #aaa;
    height: 90vh;
    padding: 6px;
`;

const Item = styled.div`
    background: #eee;
    border: 1px dashed #aaa;
    height: 100%;
`;

const verticalOptions = ['top', 'bottom', 'center', 'stretch'];
const horizontalOptions = ['left', 'right', 'center', 'stretch'];

const items:Array<JSX.Element> = [];

for (let i = 0; i < 12; i++) {
    items.push(
        <Spacer key={i} top={6} right={6} bottom={6} left={6}>
            <Item>
                <Spacer offsetType="inner" top={48} right={48} bottom={48} left={48}>
                    <Text>Item</Text>
                </Spacer>
            </Item>
        </Spacer>
    );
}

storiesOf('Box', module)
    .add('Default', () => (
        <Wrapper>
            <Box
                bleed={[select('bleed', [0, 6, 12], 0)]}
                horizontal={select('horizontal', horizontalOptions)}
                vertical={select('vertical', verticalOptions)}
                wrap={boolean('wrap', true)}
            >
                {items}
            </Box>
        </Wrapper>
    ));
