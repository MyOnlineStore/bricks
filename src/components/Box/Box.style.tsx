import { StyledComponentClass as _S } from 'styled-components';
import styled from '../../utility/styled';
import Box from './Box.template';
import formatOffset from './formatters/formatOffset';

const StyledBox = styled(Box)`
    display: flex;
    height: 100%;
    flex-wrap: ${({ wrap }):string => wrap ? 'wrap' : 'no-wrap'};
    flex-direction: row;
    justify-content: ${({ horizontal }):string => {
        switch (horizontal) {
            case undefined:
            case 'left': return 'flex-start';
            case 'right': return 'flex-end';
            default: return horizontal;
        }
    }};
    align-items: ${({ vertical }):string => {
        switch (vertical) {
            case undefined:
            case 'top': return 'flex-start';
            case 'bottom': return 'flex-end';
            default: return vertical;
        }
    }};
    align-content: ${({ vertical }):string => {
        switch (vertical) {
            case undefined:
            case 'top': return 'flex-start';
            case 'bottom': return 'flex-end';
            default: return vertical;
        }
    }};
    margin: ${({ bleed }):string => bleed !== undefined ? formatOffset(bleed) : '0'};

    > * {
        flex-grow: ${({ horizontal }):number => horizontal === 'stretch' ? 1 : 0};
        flex-shrink: 1;
    }
`;

export default StyledBox;
