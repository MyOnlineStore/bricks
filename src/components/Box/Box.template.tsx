import React, { StatelessComponent } from 'react';
import { OffsetType } from '../../types/OffsetType';
import { StyledType } from '../../utility/styled';

type PropsType = StyledType & {
    vertical?:'top'|'bottom'|'center'|'stretch';
    horizontal?:'left'|'right'|'center'|'stretch';
    wrap?:boolean;
    bleed?:Array<OffsetType>;
};

const Box:StatelessComponent<PropsType> = (props):JSX.Element => {
    return <div className={props.className}>{props.children}</div>;
};

export { PropsType };
export default Box;
