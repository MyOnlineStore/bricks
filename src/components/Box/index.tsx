import React, { FunctionComponent } from 'react';
import trbl, { TrblType } from '../../utility/_trbl';
import { StyledDiv, StyledSpan } from './style';
import { OffsetShorthandType } from '../../types/OffsetType';

type PropsType = JSX.IntrinsicElements['div'] & {
    justifyContent?:
        | 'flex-start'
        | 'flex-end'
        | 'center'
        | 'stretch'
        | 'space-between'
        | 'space-around'
        | 'space-evenly';
    alignItems?: 'flex-start' | 'flex-end' | 'center' | 'stretch';
    alignContent?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'space-between' | 'space-around';
    direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
    inline?: boolean;
    height?: string;
    width?: string;
    margin?: TrblType | OffsetShorthandType;
    padding?: TrblType | OffsetShorthandType;
    maxHeight?: string;
    minHeight?: string;
    maxWidth?: string;
    minWidth?: string;
    wrap?: boolean;
    grow?: number;
    shrink?: number;
    basis?: string;
    order?: number;
    alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
    position?: 'static' | 'relative' | 'fixed' | 'absolute';
    top?: string;
    right?: string;
    bottom?: string;
    left?: string;
};

const Box: FunctionComponent<PropsType> = (props): JSX.Element => {
    const {
        order,
        direction,
        wrap,
        height,
        width,
        maxHeight,
        minHeight,
        maxWidth,
        minWidth,
        ref,
        margin,
        padding,
        ...filteredProps
    } = props;

    const shorthandMargin = Array.isArray(margin) ? trbl(...margin) : margin;
    const shorthandPadding = Array.isArray(padding) ? trbl(...padding) : padding;

    const newProps = {
        ...filteredProps,
        flexWrap: wrap,
        elementHeight: height,
        elementWidth: width,
        elementMaxHeight: maxHeight,
        elementMinHeight: minHeight,
        elementMaxWidth: maxWidth,
        elementMinWidth: minWidth,
        flexDirection: direction,
        flexOrder: order,
        margin: shorthandMargin,
        padding: shorthandPadding,
    };

    return props.inline ? (
        <StyledSpan {...newProps}>{props.children}</StyledSpan>
    ) : (
        <StyledDiv {...newProps}>{props.children}</StyledDiv>
    );
};

export { PropsType };
export default Box;
