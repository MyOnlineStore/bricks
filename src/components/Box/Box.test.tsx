import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Box from '.';

describe('Box', () => {
    it('aligns children to the right', () => {
        const component = shallow(<Box horizontal="right" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children in the center', () => {
        const component = shallow(<Box horizontal="center" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children to the bottom', () => {
        const component = shallow(<Box vertical="bottom" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('aligns children in the vertical center', () => {
        const component = shallow(<Box vertical="center" />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('can have a bleed to compensate for the margin of children', () => {
        const component = shallow(<Box bleed={[24, 0, 'auto']} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('prevents wrapping of children', () => {
        const component = shallow(<Box wrap={false} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('stretches the children', () => {
        const component = shallow(<Box horizontal="stretch" vertical="stretch" />);

        expect(toJson(component)).toMatchSnapshot();
    });
});
