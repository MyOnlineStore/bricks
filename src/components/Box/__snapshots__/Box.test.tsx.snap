// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box aligns children in the center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  horizontal="center"
/>
`;

exports[`Box aligns children in the vertical center 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  vertical="center"
/>
`;

exports[`Box aligns children to the bottom 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  vertical="bottom"
/>
`;

exports[`Box aligns children to the right 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  horizontal="right"
/>
`;

exports[`Box can have a bleed to compensate for the margin of children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  margin: -24px -0px auto;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  bleed={
    Array [
      24,
      0,
      "auto",
    ]
  }
  className="c0"
/>
`;

exports[`Box prevents wrapping of children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  wrap={false}
/>
`;

exports[`Box stretches the children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: stretch;
  -webkit-justify-content: stretch;
  -ms-flex-pack: stretch;
  justify-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  margin: 0;
}

.c0 > * {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<Box
  className="c0"
  horizontal="stretch"
  vertical="stretch"
/>
`;
